bundles/net.tourbook/src/net/tourbook/ui/views/TourBlogView.java

```diff
@@ -240,74 +234,75 @@ public void propertyChange(final PropertyChangeEvent event) {
     */
    private void addSelectionListener() {
 
-      _postSelectionListener = new ISelectionListener() {
-         @Override
-         public void selectionChanged(final IWorkbenchPart part, final ISelection selection) {
-            if (part == TourBlogView.this) {
-               return;
-            }
-            onSelectionChanged(selection);
+      _postSelectionListener = (workbenchPart, selection) -> {
+         if (workbenchPart == TourBlogView.this) {
+            return;
          }
+         onSelectionChanged(selection);
       };
       getSite().getPage().addPostSelectionListener(_postSelectionListener);
    }
 
    private void addTourEventListener() {
 
-      _tourEventListener = new ITourEventListener() {
-         @Override
-         public void tourChanged(final IWorkbenchPart part, final TourEventId eventId, final Object eventData) {
+      _tourEventListener = (workbenchPart, eventId, eventData) -> {
 
-            if (part == TourBlogView.this) {
-               return;
-            }
+         if (workbenchPart == TourBlogView.this) {
+            return;
+         }
 
-            if ((eventId == TourEventId.TOUR_CHANGED) && (eventData instanceof TourEvent)) {
+         if ((eventId == TourEventId.TOUR_CHANGED) && (eventData instanceof TourEvent)) {
 
-               final ArrayList<TourData> modifiedTours = ((TourEvent) eventData).getModifiedTours();
-               if (modifiedTours != null) {
+            final ArrayList<TourData> modifiedTours = ((TourEvent) eventData).getModifiedTours();
+            if (modifiedTours != null) {
 
-                  // update modified tour
+               // update modified tour
 
-                  final long viewTourId = _tourData.getTourId();
+               //to reproduce and fix
+               //todo fb
 
-                  for (final TourData tourData : modifiedTours) {
-                     if (tourData.getTourId() == viewTourId) {
+               if (_tourData == null) {
```

Comment:
Possible fix for [this issue](https://github.com/mytourbook/mytourbook/issues/990)